rules/copy_file.bzl

```diff
@@ -0,0 +1,32 @@
+# Copyright 2019 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""A rule that copies a file to another place.
+
+native.genrule() is sometimes used to copy files (often wishing to rename them).
+The 'copy_file' and 'copy_xfile' rules do this with a simpler interface than
+genrule.
+
+These rules use a Bash command on Linux/macOS/non-Windows, and use a cmd.exe
+command on Windows (no Bash is required).
+"""
+
+load(
+    ":copy_file_private.bzl",
+    _copy_file = "copy_file",
+    _copy_xfile = "copy_xfile",
+)
+
+copy_file = _copy_file
+copy_xfile = _copy_xfile
```

Comment:
Drive by - why create two rules vs. one rule with an argument to say if it should be executable?  seems like an `executable` or `make_executable` argument would be more self descriptive than _xfile_ as that's a bash-ism, no?

Reply:
See the same question here: https://github.com/bazelbuild/bazel-skylib/pull/122#issuecomment-470523889

Reply:
Making a rule executable allows the user to do `bazel run`. Is it useful here?

Reply:
Yes, but more than that: it allows putting the rule into sh_binary.srcs and py_test.srcs.